### PR TITLE
HttpResponse improvements

### PR DESCRIPTION
**Pull request by BurnzZ**
Attempts to improve the docs and tests as noted down in the **TODO** section of https://github.com/scrapinghub/web-poet/pull/30.

### CODE REVIEW

File: docs/intro/from-ground-up.rst [hunk 1]
```diff
@@ -160,12 +160,12 @@ The same, but using web-poet
     # === Framework-specific I/O code
     def download_sync(url):
         resp = requests.get(url)
-        return ResponseData(url=resp.url, html=resp.text)
+        return HttpResponse(url=resp.url, body=resp.content)
```
Comment (kmike) [inline docs/intro/from-ground-up.rst:163]: this needs to be a bit more elaborate, we need to pass headers as well, in order for the response to be decoded properly

File: docs/intro/from-ground-up.rst [hunk 2]
```diff
@@ -160,12 +160,12 @@ The same, but using web-poet
...
 
     async def download_async(session, url):
         async with session.get(url) as resp:
-            text = await resp.text()
-        return ResponseData(url=resp.url, html=text)
+            body = await resp.body
+        return HttpResponse(url=resp.url, body=body)
```
Comment (kmike) [inline docs/intro/from-ground-up.rst:168]: we need to pass headers here too

File: docs/intro/from-ground-up.rst (hunk 1 above)
Comment (kmike) [reply docs/intro/from-ground-up.rst:163]: See also: https://github.com/scrapinghub/web-poet/commit/92f114cd1c7294e18f70ef76f2b670a74bdb3982 (the commit is in a branch, it doesn't belong to any PR)

File: tests/test_page_inputs.py [hunk 3]
```diff
@@ -117,3 +117,30 @@ def test_http_response_headers_init_invalid():
...
+def test_http_response_selectors(book_list_html_response):
+    title = "All products | Books to Scrape - Sandbox"
+
+    assert title == book_list_html_response.css("title ::text").get("").strip()
+    assert title == book_list_html_response.xpath("//title/text()").get("").strip()
+
+
+def test_http_response_json():
+    http_body = HttpResponseBody(b'{"key": "value"}')
+    response = HttpResponse("http://example.com", body=http_body)
```
Comment (kmike) [inline tests/test_page_inputs.py:131]: FTR, wrappinig body in HttpResponseBody is not really required, you can pass bytes as well

File: tests/test_page_inputs.py [hunk 4]
```diff
@@ -117,3 +117,30 @@ def test_http_response_headers_init_invalid():
...
+
+    assert response.json() == {"key": "value"}
```
Comment (kmike) [inline tests/test_page_inputs.py:133]: For response.json we may check a couple more things, similar to test_http_response_body_json

File: docs/intro/from-ground-up.rst [hunk 5]
```diff
@@ -178,9 +178,9 @@ The same, but using web-poet
 
 Differences from a previous example:
 
-* instead of dicts with "url" and "text" fields, :class:`~.ResponseData`
-  instances are used. :class:`~.ResponseData` is a simple structure with
-  two fields ("url" and "html"), defined by web-poet;
+* instead of dicts with "url" and "text" fields, :class:`~.HttpResponse`
+  instances are used. :class:`~.HttpResponse` is a simple structure with
+  two fields ("url" and "body"), defined by web-poet;
```
Comment (kmike) [inline docs/intro/from-ground-up.rst:183]: It has more fields now. Wee should check the whole from-ground-up tutorial; it has some inconsistencies after the removal of ResponseData.

File: docs/intro/from-ground-up.rst (hunk 1 above)
Comment (BurnzZ) [reply docs/intro/from-ground-up.rst]: Done! https://github.com/scrapinghub/web-poet/pull/31/commits/7125ec7b8a027a37d58242847de975b14ff0839f

File: docs/intro/from-ground-up.rst (hunk 2 above)
Comment (BurnzZ) [reply docs/intro/from-ground-up.rst]: Done. https://github.com/scrapinghub/web-poet/pull/31/commits/7125ec7b8a027a37d58242847de975b14ff0839f

File: tests/test_page_inputs.py (hunk 4 above)
Comment (BurnzZ) [reply tests/test_page_inputs.py:137]: Good call. Added them in https://github.com/scrapinghub/web-poet/pull/31/commits/3aed4915fc1c44dee8c15349a7055b5c25ebd197.

File: tests/test_page_inputs.py (hunk 3 above)
Comment (BurnzZ) [reply tests/test_page_inputs.py]: Updated this in https://github.com/scrapinghub/web-poet/pull/31/commits/3aed4915fc1c44dee8c15349a7055b5c25ebd197.

**Comment by kmike** [comment]
Thanks @BurnzZ for the help!

**Pull request merged**